tests/e2e/e2e_test.go

```diff
@@ -3034,5 +3034,149 @@ func TestE2E(t *testing.T) {
 
 			},
 		},
+
+		testExecution{
+			Name:       "TestKubeletConfigRemediation",
+			IsParallel: false,
+			TestFn: func(t *testing.T, f *framework.Framework, ctx *framework.Context, mcTctx *mcTestCtx, namespace string) error {
+
+				const baselineImage = "quay.io/wenshen/co_content@sha256:2c16e3d4732bdb2e87ef2833d2f1368295d96f134df805b34710cfeb62d1d028"
+				const requiredRule = "kubelet-eviction-thresholds-set-hard-imagefs-available"
+				pbName := getObjNameFromTest(t)
+				prefixName := func(profName, ruleBaseName string) string { return profName + "-" + ruleBaseName }
+
+				ocpPb := &compv1alpha1.ProfileBundle{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      pbName,
+						Namespace: namespace,
+					},
+					Spec: compv1alpha1.ProfileBundleSpec{
+						ContentImage: baselineImage,
+						ContentFile:  ocpContentFile,
+					},
+				}
+				if err := f.Client.Create(goctx.TODO(), ocpPb, getCleanupOpts(ctx)); err != nil {
+					return err
+				}
+				if err := waitForProfileBundleStatus(t, f, namespace, pbName, compv1alpha1.DataStreamValid); err != nil {
+					return err
+				}
+
+				// Check that if the rule we are going to test is there
+				err, found := doesRuleExist(f, ocpPb.Namespace, prefixName(pbName, requiredRule))
+				if err != nil {
+					return err
+				} else if found != true {
+					E2EErrorf(t, "Expected rule %s not found", prefixName(pbName, requiredRule))
+					return err
+				}
+
+				suiteName := "kubeletconfig-test-node"
+				scanName := "kubeletconfig-test-node"
+
+				tp := &compv1alpha1.TailoredProfile{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      suiteName,
+						Namespace: namespace,
+					},
+					Spec: compv1alpha1.TailoredProfileSpec{
+						Title:       "kubeletconfig-Test",
+						Description: "A test tailored profile to kubeletconfig remediation",
+						EnableRules: []compv1alpha1.RuleReferenceSpec{
+							{
+								Name:      prefixName(pbName, requiredRule),
+								Rationale: "To be tested",
+							},
+						},
+						SetValues: []compv1alpha1.VariableValueSpec{
+							{
+								Name:      prefixName(pbName, "var-kubelet-evictionhard-imagefs-available"),
+								Rationale: "Value to be set",
+								Value:     "20%",
+							},
+						},
+					},
+				}
+				mcTctx.ensureE2EPool()
```

Comment:
This creates a custom e2e pool. We use this so we have shorter MCO update times. It's an e2e pool with one node only. If you start going through that function and the ones it calls, you'll notice it creates a MachineConfigPool that tracks only ONE node, this way, we keep it small and fast.